config/jobs/kubernetes/cloud-provider-gcp/cloud-provider-gcp-periodics.yaml

```diff
@@ -375,5 +375,10 @@ periodics:
       args:
       - "/bin/bash"
       - "-c"
-      - cd /workspace/cloud-provider-gcp;
+      - set -o errexit;
+        set -o nounset;
+        set -o pipefail;
+        set -o xtrace;
+        cd $GOPATH/src/cloud-provider-gcp;
+        e2e/setup-kube.sh;
```

Comment:
setup-kube.sh currently does two things:

1) clone kubernetes into ../kubernetes
2) set up go.work etc to build with it


Maybe it would be better if we actually just cloned kubernetes as another `extra_refs`, and passed an env var to the test to specify that we want to build with that version of k8s?

Reply:
yes, it was that way before, but I changed because I thought it was causing issues https://github.com/kubernetes/test-infra/pull/28764 but it turned out that problem was not that

Reply:
I mean, we should use `extra_refs`

Reply:
can you add it ?

Reply:
Yes, sgtm - will do!

Reply:
Done - still some iteration possible e.g. should we create the workspace for cloud-provider-gcp separately from adding kubernetes to it, but I think it's a start and we probably want to get it working first!